### PR TITLE
Search: update notice box styling for Record Meter

### PR DESCRIPTION
**Pull request by annacmc**
This PR updates Record Meter's notice box to include a header text field, along with styling to make the header text field bold on error/alert messages, and with correct padding. 

This also fixes some requested padding changes, and resolves the `learn more` link disappearing on hover issue. 

This is how each of the notices now look:

Screen Shot 2022-06-01 at 14 39 19

Screen Shot 2022-06-01 at 14 30 31

Screen Shot 2022-06-01 at 14 33 44

Screen Shot 2022-06-01 at 14 34 14

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Go to` /wp-admin/admin.php?page=jetpack-search&features=record-meter` on your test site with Search enabled.
The easiest way to trigger this behaviour is to edit 
* Trigger the different notice box behaviours by toggling off/on the different states `hasBeenIndexed` `hasValidData` and 'hasItems` to trigger the different notices. Note that `hasValidData` being false should resolve a red notice box with a bolded header text.
* It is also possible to trigger the 'close to limit' notice by adjusting the record count. 
* Check that the 'learn more link' (when displayed) remains black and visible on hover

| toggle the states in inspector  | red noticed when `hasValidData` is false |
| ------------- | ------------- |
| image  | image  |

### CODE REVIEW

File: projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx:133]: If this is a heading it feels like we should mark it up as such.

In our page it's a `h3` ('Your search records' is h2).

Comment (bluefuton) [reply projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx:133]:
```suggestion
			{ notice.header && <h3 className="dops-notice__header">{ notice.header }</h3> }
```

Comment (bluefuton) [reply projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx:133]: (This gets more interesting in reusable components - for example if we were adding a heading to SimpleNotice itself! Some discussion here: https://www.a11yproject.com/posts/how-to-accessible-heading-structure/#headings-in-reusable-components)

File: projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx:49]: In the designs, the message body always seems to end with a full stop.
```suggestion
					"Once you hit %s indexed records, you'll automatically be billed for the next tier.",
```

**Review by bluefuton — DISMISSED** [review]
A few small things to address:

- Header styles need adjusting now we're using a `<h3>` (sorry, this is my fault)
- The designs use "Learn more" followed by a full stop for link text rather than "learn more".
- Let's double check the spacing. I think the info icon and close button need to come down a bit further (24px).

Screen Shot 2022-05-18 at 19 43 14

**Comment by sdixon194** [comment]
@bluefuton I think this is fixed up now! Tldr, seems somehow this PR ended up with a bunch of changelog files for dependency updates that weren't related to this PR. CI saw the changelog entries and assumed all of those projects needed a version bump.

I just removed the errant changelog entries, merged with master, and updated the search project version since the atomic release today. I think this PR is good to go, but let us know if you run into any other issues with it 👍

**Comment by bluefuton** [comment]
Excellent! Thanks for your help @sdixon194 :)

**Comment by annacmc** [comment]
> A few small things to address:
> 
> * Header styles need adjusting now we're using a `<h3>` (sorry, this is my fault)
> * The designs use "Learn more" followed by a full stop for link text rather than "learn more".
> * Let's double check the spacing. I think the info icon and close button need to come down a bit further (24px).

Okay @bluefuton these should all be addressed now, 🤞 ready for another look I think.

**Comment by bluefuton** [comment]
Thanks @annacmc! I think we still have a few minor adjustments to make. Here's the design:

168988808-463044fc-e9c9-4e73-92bb-f31d5f552568

...and here's how it currently renders in Firefox:

Screen Shot 2022-05-26 at 12 20 52

The things I can spot are:
- [ ] the spacing is too great between the left and right edge and the icons (30px+ rather than 24px)
- [ ] the line height looks too high (quite a bit space between the two lines of text)
- [ ] the text seems to wrap to the second line too early
- [ ] tiny one - in the design, the full stop is not linked in "Learn more."

**Comment by annacmc** [comment]
Thanks for taking a look! looks like I should be a little more thorough in testing in FF as it behaves different. 

> the spacing is too great between the left and right edge and the icons (30px+ rather than 24px)

Nice catch. This looks like something to do with how firefox vs chrome handles flexboxes 🤔  looking into it. 

> the line height looks too high (quite a bit space between the two lines of text)
> the text seems to wrap to the second line too early
> tiny one - in the design, the full stop is not linked in "Learn more."

**Comment by bluefuton** [comment]
> something to do with how firefox vs chrome handles flexboxes

☹️ Thanks for investigating!

> I could certainly append an un-linked fullstop after the line using CSS though. Do you think that would be acceptable?

I think it's safest to have the component support it (maybe `link.text_after_link`?) because not all languages use a full stop: https://www.quora.com/Do-all-living-languages-use-the-period-to-end-a-sentence.

**Comment by bluefuton** [comment]
Having played with this a bit I made some slight adjustments to the layout and copy:
* Move 'Learn more' call to action to it's own line (this helps on mobile especially)
* Added more "why" text to explain the different situations

This is how each of the notices now look:

Screen Shot 2022-06-01 at 14 39 19

Screen Shot 2022-06-01 at 14 30 31

Screen Shot 2022-06-01 at 14 33 44

Screen Shot 2022-06-01 at 14 34 14

File: projects/js-packages/components/changelog/add-info-outline-gridicon
Comment (jsnmoon) [inline projects/js-packages/components/changelog/add-info-outline-gridicon:4]: This looks like an extraneous changelog? I don't see any modification to the components package in this PR.

This changelog is causing a minor version bump for `jetpack-components` and therefore triggering a cascade of version bumps and changelogs across the repository.

File: projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx
Comment (jsnmoon) [inline projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx:109]: Why are we removing the notice for going over the record limit?

File: projects/js-packages/components/changelog/add-info-outline-gridicon
Comment (bluefuton) [reply projects/js-packages/components/changelog/add-info-outline-gridicon:4]: Ah. I wondered why fixup-project-versions created such a cascade of files.

The changelog must be left over from a previous rebase. I'll remove it 👍

File: projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx
Comment (jsnmoon) [inline projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx:108]: Is this addition intentional? Wouldn't this line lead to always showing the near-the-limit notice?

Comment (bluefuton) [reply projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx:108]: Excellent spot! This is left over from my testing. I'll remove it 👍

**Comment by kangzj** [comment]
Hi @bluefuton, I had a go, it works well. However the JS tests might need to be fixed before merging - seems only a wording difference.

>   [test-scripts]     > 62 | 		expect( screen.getByText( /close to the max amount of records/i ) ).toBeVisible();

File: projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx
Comment (kangzj) [inline projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx:41]: should the id be '4'?

Comment (bluefuton) [reply projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx:41]: Indeed it should. Good spot 👍 

I suspect the `id` key isn't actually used.

**Review by kangzj — APPROVED** [review]
LGTM

**Pull request merged**
